src/main/java/com/rsr5/gocd/riemann_notifier/PipelineDetailsPopulator.java

```diff
@@ -26,11 +41,42 @@ private JsonObject mergeInPipelineInstanceDetails(String attributeName,
         return json;
     }
 
-    protected JsonElement downloadPipelineDetails(String pipelineName) throws IOException {
-        String path = "/go/api/pipelines/" + pipelineName + "/history";
-        JsonObject json = accessor.get(path).getAsJsonObject();
-        JsonArray pipelines = json.get("pipelines").getAsJsonArray();
-        return pipelines.get(0);
+    /**
+     * Performs a depth-first search of the artifacts, returning a JsonObject
+     * translation of the first artifact named "results.xml".
+     */
+    protected JsonElement findJunitResults(JsonArray artifacts) throws IOException {
+        for (int i = 0; i < artifacts.size(); i++) {
+            JsonObject artifact = artifacts.get(i).getAsJsonObject();
+            if ("folder".equals(artifact.get("type").getAsString())) {
+                JsonElement result = findJunitResults(artifact.get("files").getAsJsonArray());
+                if (null != result) {
+                    return result;
+                }
+            } else if ("file".equals(artifact.get("type").getAsString()) &&
+                       "results.xml".equals(artifact.get("name").getAsString())) {
+                URL url = new URL(artifact.get("url").getAsString());
+                HttpURLConnection con = accessor.getConnection(url.getPath());
+                con.connect();
+                if (con.getResponseCode() != 200) {
+                    LOGGER.error("Server returned error when trying to get JUnit results: " + con.getResponseMessage());
+                    return null;
+                }
+                BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+                String inputLine;
+                StringBuffer response = new StringBuffer();
+
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                in.close();
```

Comment:
This would benefit from using Java's [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) statement (like Clojure's `with-open`):
```suggestion
                try (BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()))) {
                  String inputLine;
                  StringBuffer response = new StringBuffer();

                  while ((inputLine = in.readLine()) != null) {
                      response.append(inputLine);
                  }
                }
```